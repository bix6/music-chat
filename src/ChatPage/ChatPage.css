.chat-page-main {
  margin: 0 auto;
  max-width: 700px;
  height: 100%;
  border: 3px solid #43cbff;
  background-image: linear-gradient(20deg, #9708cc 10%, #43cbff 100%);
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 0.2rem 0.2rem 0.2rem;
}

.chat-selector-form {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  flex: 0 1 auto;
  width: 100%;
}

.chat-selector-form label {
  flex: 0 1 auto;
}

.chat-selector-form select {
  flex: 1 1 auto;
  text-align: center;
  margin: 0 2rem 0 0.2rem;
}

.chat-selector-form button {
}

.chat-selector-form p {
  text-align: center;
  flex: 1 1 100%;
  margin: 1rem 0;
  background-color: #9708cc;
  color: #44ffd2;
  position: relative;
}

.chat-convo-div {
  flex: 10 1 auto;
  width: 100%;
}

.chat-input-div {
  flex: 0 1 auto;
  width: 100%;
}

.chat-input-form {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
}

.chat-input-flex-div {
  display: flex;
  align-items: center;
  width: 100%;
}

.chat-input-form label {
  flex: 1 1 auto;
}

.chat-input-form input {
  flex: 100 1 auto;
}

.chat-input-form button {
  flex: 1 1 auto;
}

.chat-input-form .display-message {
  flex: 1 1 100%;
}

iframe {
  width: 100%;
}
