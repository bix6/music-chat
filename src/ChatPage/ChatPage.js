import React from 'react';
import HeaderComponent from '../shared-components/HeaderComponent/HeaderComponent';
import ChatSelector from './ChatSelector/ChatSelector';
import ChatConvo from './ChatConvo/ChatConvo';
import ChatInput from './ChatInput/ChatInput';
import SearchResults from './SearchResults/SearchResults';

class ChatPage extends React.Component {
    state = {
        chatroomList: [
            {
                name: 'Global',
                description: 'Welcome to the World Wide Music Chat!'
            },
            {
                name: 'Playlists',
                description: 'A place to discuss your favorite playlists.'
            },
            {
                name: 'Mood',
                description: 'How do you feel today?'
            },
            {
                name: 'Artists',
                description: 'Keep up to date on your favorite artists.'
            },
            {
                name: 'Songs',
                description: 'All the single songs'
            },
            {
                name: 'Doin\' Time',
                description: 'Chat about Doin\' time by Sublime (and now Lana)'
            }
        ],
        currentChatroom: 'Global',
        messages: {
            'Global': [
                'Jack: Music Chat Baby!!!!!',
                'Jac: Just gonna talk to myself',
                'Ja: A cacophony of Jacks',
                'J: Phony, real phony'
            ],
            'Playlists': [
                'Dalbert: Please let me share playlists',
                'Jack: Ok',
                'Dalbert: Please send Lorde to my estate',
                'Jack: She will be escorted by 20 Golden Kiwis'
            ],
            'Mood': [
                'Brownie: My mood is sassy cuz it is hot and I\'m listening to Milkshake',
                'Basil: I am so chilled out listening to Baby Take Off Your Cool',
                'Brownie: Damn that\'s fresh Basil'
            ],
            'Artists': [
                'Jerry: Have you heard of Timbo?',
                'Timbo: Yes',
                'Jerry: He is really good',
                'Timbo: No'
            ],
            'Songs': [
                'Chester: I wrote a song',
                'Hubert: About what?',
                'Chester: You, Hubert. It\'s about you Hubert.',
            ],
            'Doin\' Time': [
                'Nick: Alright! Now we can chat about individual songs.',
                'Jack: Summertime and the coding is not easy.',
                'Nick: Ain\'t no thang.',
                'Jack: So which version would you pick: Sublime or Lana?',
                'Nick: ...'

            ]
        },
        searchResults: null
    }

    updateCurrentChatroom = (chatroom) => {
        this.setState({ currentChatroom: chatroom });
    }

    sendMessage = (messageIn) => {
        let message = this.props.username + ': ' + messageIn;
        let messages = this.state.messages;
        messages[this.state.currentChatroom].push(message);
        this.setState({
            messages: messages
        })
    }

    setSearchResults = (searchResults) => {
        this.setState({ searchResults: searchResults})
    }

    embedVideo = (index) => {
        const videoId = this.state.searchResults[index].videoId;
        this.sendMessage(`YOUTUBE_IFRAME ${ videoId }`)
        this.setState({ searchResults: null})
    }

    render() {
        return (
            <>
                <HeaderComponent />
                <main>
                    <ChatSelector 
                        chatroomList={ this.state.chatroomList }
                        currentChatroom={ this.state.currentChatroom }
                        updateCurrentChatroom={ this.updateCurrentChatroom }/>
                    <ChatConvo 
                        currentChatroom={ this.state.currentChatroom }
                        messages={ this.state.messages }/>
                    <ChatInput 
                        sendMessage={ this.sendMessage }
                        setSearchResults={ this.setSearchResults }/>
                    { this.state.searchResults 
                        && <SearchResults 
                            searchResults={ this.state.searchResults }
                            embedVideo={ this.embedVideo }/> }
                </main>
            </>
        )
    }
}

export default ChatPage;