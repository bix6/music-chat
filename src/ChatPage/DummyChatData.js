const chatroomList = [
  {
    name: "Global",
    description: "Welcome to the World Wide Music Chat!",
  },
  {
    name: "Playlists",
    description: "A place to discuss your favorite playlists.",
  },
  {
    name: "Mood",
    description: "How do you feel today?",
  },
  {
    name: "Artists",
    description: "Keep up to date on your favorite artists.",
  },
  {
    name: "Songs",
    description: "All the single songs",
  },
  {
    name: "Doin' Time",
    description: "Chat about Doin' Time by Sublime (and now Lana)",
  },
];

const messages = {
  Global: [
    {
      username: "Alicia",
      contentType: "text",
      message: "Music Chat, Baby.",
      contentId: null,
    },
    {
      username: "Jack",
      contentType: "text",
      message: "You Don't Know My Name.",
      contentId: null,
    },
    {
      username: "Alicia",
      contentType: "text",
      message: "But you're Unbreakable.",
      contentId: null,
    },
    {
      username: "Jack",
      contentType: "text",
      message: "If I Ain't Got You, who would I be?",
      contentId: null,
    },
    {
      username: "Alicia",
      contentType: "youtube video",
      message: null,
      contentId: "rywUS-ohqeE",
    },
  ],
  Playlists: [
    {
      username: "Dalbert",
      contentType: "text",
      message: "Let's share playlists",
      contentId: null,
    },
    {
      username: "Jack",
      contentType: "text",
      message: "Ok",
      contentId: null,
    },
    {
      username: "Dalbert",
      contentType: "text",
      message: "Please send Lorde to my estate",
      contentId: null,
    },
    {
      username: "Jack",
      contentType: "text",
      message: "She'll be escorted by 20 Golden Kiwis",
      contentId: null,
    },
  ],
  Mood: [
    {
      username: "Brownie",
      contentType: "text",
      message: "My mood is sassy cuz it's hot and I'm listening to Milkshake",
      contentId: null,
    },
    {
      username: "Basil",
      contentType: "text",
      message: "I'm so chilled out listening to Baby Take Off Your Cool",
      contentId: null,
    },
    {
      username: "Brownie",
      contentType: "text",
      message: "Damn, that's fresh Basil",
      contentId: null,
    },
  ],
  Artists: [
    {
      username: "Jerry",
      contentType: "text",
      message: "Have you heard of Timbo?",
      contentId: null,
    },
    {
      username: "Timbo",
      contentType: "text",
      message: "Yes",
      contentId: null,
    },
    {
      username: "Jerry",
      contentType: "text",
      message: "He's really good",
      contentId: null,
    },
    {
      username: "Timbo",
      contentType: "text",
      message: "No",
      contentId: null,
    },
  ],
  Songs: [
    {
      username: "Chester",
      contentType: "text",
      message: "I wrote a song",
      contentId: null,
    },
    {
      username: "Hubert",
      contentType: "text",
      message: "About what?",
      contentId: null,
    },
    {
      username: "Chester",
      contentType: "text",
      message: "It's about you Hubert",
      contentId: null,
    },
  ],
  "Doin' Time": [
    {
      username: "Nick",
      contentType: "text",
      message: "Alright we got songs.",
      contentId: null,
    },
    {
      username: "Jack",
      contentType: "text",
      message: "Summertime and the coding is not easy",
      contentId: null,
    },
    {
      username: "Nick",
      contentType: "text",
      message: "Ain't no thang...",
      contentId: null,
    },
  ],
};

export { chatroomList, messages };
