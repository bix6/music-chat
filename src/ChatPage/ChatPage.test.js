import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import ChatPage from './ChatPage';

describe('ChatPage Component', () => {
    it('Smoke Test: Renders Default', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <ChatPage />
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('Snapshot Test: Default', () => {
        const wrapper = shallow(<ChatPage />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})