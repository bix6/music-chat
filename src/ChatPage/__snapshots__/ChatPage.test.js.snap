// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatPage Component Snapshot Test: Default 1`] = `
<Fragment>
  <HeaderComponent />
  <main>
    <ChatSelector
      chatroomList={
        Array [
          Object {
            "description": "Welcome to the World Wide Music Chat!",
            "name": "Global",
          },
          Object {
            "description": "A place to discuss your favorite playlists.",
            "name": "Playlists",
          },
          Object {
            "description": "How do you feel today?",
            "name": "Mood",
          },
          Object {
            "description": "Keep up to date on your favorite artists.",
            "name": "Artists",
          },
          Object {
            "description": "All the single songs",
            "name": "Songs",
          },
          Object {
            "description": "Chat about Doin' Time by Sublime (and now Lana)",
            "name": "Doin' Time",
          },
        ]
      }
      currentChatroom="Global"
      updateCurrentChatroom={[Function]}
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Create Chatroom
    </button>
    <NewChatroom
      closeNewChatroom={[Function]}
      createChatroom={[Function]}
      newChatroomDisplay="none"
    />
    <ChatConvo
      currentChatroom="Global"
      messages={
        Object {
          "Artists": Array [
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "Have you heard of Timbo?",
              "username": "Jerry",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "Yes",
              "username": "Timbo",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "He is really good",
              "username": "Jerry",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "No",
              "username": "Timbo",
            },
          ],
          "Doin' Time": Array [
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "Alright! Now we can chat about individual songs.",
              "username": "Nick",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "Summertime and the coding is not easy",
              "username": "Jack",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "Ain't no thang",
              "username": "Nick",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "So which version would you pick: Sublime or Lana?",
              "username": "Jack",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "...",
              "username": "Nick",
            },
          ],
          "Global": Array [
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "Music Chat Baby!!!!!",
              "username": "Alicia",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "You Don't Know My Name.",
              "username": "Jack",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "But you're Unbreakable.",
              "username": "Alicia",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "If I Ain't Got You, who would I be?",
              "username": "Jack",
            },
            Object {
              "contentId": "rywUS-ohqeE",
              "contentType": "youtube video",
              "message": null,
              "username": "Alicia",
            },
          ],
          "Mood": Array [
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "My mood is sassy cuz it is hot and I'm listening to Milkshake",
              "username": "Brownie",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "I am so chilled out listening to Baby Take Off Your Cool",
              "username": "Basil",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "Damn that's fresh Basil",
              "username": "Brownie",
            },
          ],
          "Playlists": Array [
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "Please let me share playlists",
              "username": "Dalbert",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "Ok",
              "username": "Jack",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "Please send Lorde to my estate",
              "username": "Dalbert",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "She will be escorted by 20 Golden Kiwis",
              "username": "Jack",
            },
          ],
          "Songs": Array [
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "I wrote a song",
              "username": "Chester",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "About what?",
              "username": "Hubert",
            },
            Object {
              "contentId": null,
              "contentType": "text",
              "message": "You, Hubert. It's about you Hubert",
              "username": "Chester",
            },
          ],
        }
      }
    />
    <ChatInput
      sendMessage={[Function]}
      setSearchResults={[Function]}
    />
  </main>
</Fragment>
`;
